src/middlewared/middlewared/utils/prctl.py

```diff
@@ -0,0 +1,37 @@
+import contextlib
+import ctypes
+import enum
+import signal
+
+__all__ = ['set_name', 'set_pdeath_sig']
+
+
+class Prctl(enum.IntEnum):
+    # from linux/prctl.h
+    SET_PDEATHSIG = 1
+    SET_NAME = 15
+
+
+@contextlib.contextmanager
+def load_libc():
+    libc = None
+    try:
+        libc = ctypes.CDLL(ctypes.util.find_library('c'))
```

Comment:
Is this thread-safe?

Reply:
Should be, documentation states that we only need to worry about tracked references when we pass callback functions (which we're not)